library/Fission/AWS/Route53.hs

```diff
@@ -0,0 +1,73 @@
+module Fission.AWS.Route53 
+  ( createChangeRequest
+  , registerDomain
+  ) where
+
+import RIO
+import Servant
+
+import Data.Has
+import qualified Fission.Config as Config
+
+import Network.AWS.Route53 as Route53
+import Network.AWS.Prelude as AWS
+import Network.AWS.Auth    as AWS
+import Fission.AWS.Types   as AWS
+import Control.Monad.Trans.AWS
+
+import Fission.Internal.Constraint
+
+
+import Control.Lens ((?~))
+
+registerDomain :: MonadRIO       cfg m
+            => HasLogFunc cfg
+           => Has AWS.AccessKey  cfg
+           => Has AWS.SecretKey  cfg
+           => Has AWS.ZoneId     cfg
+           => RecordType
+           -> Text
+           -> Text
+           -> m(Maybe ServerError)
+registerDomain recordType domain content = do 
+  logDebug $ "DNS record at: " <> displayShow domain
+  env <- createEnv
+  req <- createChangeRequest recordType domain content
+
+  liftIO . runResourceT . runAWST env . within NorthVirginia $ do
+    res <- send req
+    return . parseStatus $ res ^. crrsrsResponseStatus
+
+createEnv :: MonadRIO           cfg m
+           => Has AWS.AccessKey cfg
+           => Has AWS.SecretKey cfg
+           => m(Env)
+createEnv = do
+  AccessKey accessKey <- Config.get
+  SecretKey secretKey <- Config.get
+  liftIO $ newEnv $ FromKeys (AccessKey accessKey) (SecretKey secretKey)
```

Comment:
Can also give a type hint to avoid unwrapping / rewrapping

```haskell
accessKey :: AccessKey <- Config.get
secretKey :: SecretKey <- Config.get
liftIO $ newEnv $ FromKeys accessKey secretKey
```